lua_examples/lfs/_init.lua

```diff
@@ -24,36 +24,45 @@
                  print(table.concat(LFS._list,'\n'))
                gives you a single column listing of all modules in the LFS.
 
+   For Lua 5.3 LFS table is already populated by the firmware itself so this part of the
+   code is skipped.
 ---------------------------------------------------------------------------------]]
 
-local index = node.flashindex
-
-local lfs_t = {
-  __index = function(_, name)
-      local fn_ut, ba, ma, size, modules = index(name)
-      if not ba then
-        return fn_ut
-      elseif name == '_time' then
-        return fn_ut
-      elseif name == '_config' then
-        local fs_ma, fs_size = file.fscfg()
-        return {lfs_base = ba, lfs_mapped = ma, lfs_size = size,
-                fs_mapped = fs_ma, fs_size = fs_size}
-      elseif name == '_list' then
-        return modules
-      else
-        return nil
-      end
-    end,
-
-  __newindex = function(_, name, value) -- luacheck: no unused
-      error("LFS is readonly. Invalid write to LFS." .. name, 2)
-    end,
-
+local G=_ENV or getfenv()
+if _VERSION == 'Lua 5.1' then 
+  local index = node.flashindex
+  local lfs_t = {
+    __index = function(_, name)
+        local fn_ut, ba, ma, size, modules = index(name)
+        if not ba then
+          return fn_ut
+        elseif name == '_time' then
+          return fn_ut
+        elseif name == '_config' then
+          local fs_ma, fs_size = file.fscfg()
+          return {lfs_base = ba, lfs_mapped = ma, lfs_size = size,
+                  fs_mapped = fs_ma, fs_size = fs_size}
+        elseif name == '_list' then
+          return modules
+        else
+          return nil
+        end
+      end,
+
+    __newindex = function(_, name, value) -- luacheck: no unused
+        error("LFS is readonly. Invalid write to LFS." .. name, 2)
+      end,
   }
 
-local G=getfenv()
-G.LFS = setmetatable(lfs_t,lfs_t)
+  G.LFS = setmetatable(lfs_t,lfs_t) 
+  --[[-------------------------------------------------------------------------------
+    You can add any other initialisation here, for example a couple of the globals
+    are never used, so setting them to nil saves a couple of global entries
+  ---------------------------------------------------------------------------------]]
```

Comment:
Split this comment in twain; this, being gated by the `_VERSION` check, is not where we want to encourage users to extend `_init.lua`, but these next lines still need some commentary.

Reply:
Do you mean to have this section also in the `else` part of the `_VERSION` check?
There's no need to nil `G.module` and `package.seeall` under Lua 5.3.
Or do we just modify the comment that it applies to Lua 5.1 solely?

Reply:
I just mean that the encouragement "You can add any other initialisation here" is now out of place.